python/test/unit/language/test_conversions.py

```diff
@@ -366,3 +366,77 @@ def test_typeconvert_downcast(src_dtype, dst_dtype, rounding, max_repr, device):
 
     for i in range(256):
         downcast_test(getattr(tl, src_dtype), getattr(tl, dst_dtype), rounding, *stuff, max_repr, i, device=device)
+
+@pytest.mark.parametrize("src_dtype, dst_dtype, rounding", [
+    ('float32', 'float8e4nv', 'rtne'),
+    ('float16', 'float8e4nv', 'rtne'),
+    ('bfloat16', 'float8e4nv', 'rtne'),
+    ('float32', 'float8e5', 'rtne'),
+    ('float16', 'float8e5', 'rtne'),
+    ('bfloat16', 'float8e5', 'rtne'),
+])
+@pytest.mark.parametrize("mode", [
+    'max', 'min', 'inf', '-inf', 'nan',
+])
+def test_typeconvert_downcast_clamping(src_dtype, dst_dtype, rounding, mode, device):
+    if src_dtype != 'float32' and torch.cuda.get_device_capability(0) < (9, 0):
+        pytest.skip("non-float32 downcast tests only supported on NVGPU with compute capability 9.0+")
+
+    if dst_dtype in ('float8e5', 'float8e4nv') and rounding == 'rtne' and torch.cuda.get_device_capability(0) < (9, 0):
+        pytest.skip(f"{dst_dtype} downcast with RTNE rounding tests only supported on NVGPU with compute capability 9.0+")
```

Comment:
Might want to put under `if is_cuda() and torch.cuda.get_device_capability(0) < (9, 0):`? For HIP `torch.cuda.get_device_capability` also returns something which doesn't quite make sense..